dotCMS/src/main/java/com/dotmarketing/quartz/DotJobStore.java

```diff
@@ -156,9 +158,16 @@ public void shutdown() {
 		super.initialize(loadHelper, signaler);
 	}
 
-	protected void closeConnection(Connection con) {
-		DbConnectionFactory.closeSilently();
-		CloseUtils.closeQuietly(con);
+	protected void closeConnection(final Connection conn) {
+	    if(DbConnectionFactory.connectionExists()) {
+	        final Connection conn2 = DbConnectionFactory.getConnection();
+	        if(conn == conn2) {
+	            Logger.warnAndDebug(this.getClass(), new DotRuntimeException("Quartz using a dotCMS connection, should not be.  Closing"));
+	            DbConnectionFactory.closeSilently();
+	        }
+	    }
+	    CloseUtils.closeQuietly(conn);

```

Comment:
what is the difference between `DbConnectionFactory.closeSilently();` and `CloseUtils.closeQuietly(conn);`? wouldn't the connection be closed anyway when the `CloseUtils.closeQuietly(conn);` is called?

Reply:
The difference is that `DbConnectionFactory.closeSilently();` does the dotCMS work - it commits transactions, closes any Hibernate session, and removes the connection from the Threadlocal connectionholder.  The CloseUtils.closeQuietly just closes whatever connection is passed to it without doing all the other work - which is what quartz is expecting to happen.